pkg/sessions/redis/redis_store.go

```diff
@@ -170,12 +171,39 @@ func buildStandaloneClient(opts options.RedisStoreOptions) (Client, error) {
 		return nil, err
 	}
 
+	if err := setupAWSIAMAuth(opts, opt); err != nil {
+		return nil, err
+	}
+
 	opt.ConnMaxIdleTime = time.Duration(opts.IdleTimeout) * time.Second
 
 	client := redis.NewClient(opt)
 	return newClient(client), nil
 }
 
+func setupAWSIAMAuth(opts options.RedisStoreOptions, opt *redis.Options) error {
+	if !opts.UseAWSIAMAuth {
+		return nil
+	}
+	if opts.AWSServiceName != "elasticache" && opts.AWSServiceName != "memorydb" {
+		return fmt.Errorf("AWS IAM auth is only supported for elasticache and memorydb")
+	}
+	generator, err := auth.New(opts.AWSServiceName, opts.AWSClusterName, opts.AWSUsername)
+	if err != nil {
+		return fmt.Errorf("error creating AWS IAM auth token generator: %v", err)
+	}
+	opt.CredentialsProvider = func() (username string, password string) {
+		token, err := generator.Generate()
+		if err != nil {
+			logger.Errorf("error generating AWS IAM auth token: %v", err)
+		}
+		return opts.AWSUsername, token
+	}
+	// AWS services has a max connection lifetime of 12 hours. This is set to 11 hours to give some buffer time
```

Comment:
I assume there's no action needed and the connection refreshes itself automatically after this period?